src/compiler/transformers/declarations.ts

```diff
@@ -265,9 +289,190 @@ export function transformDeclarations(context: TransformationContext) {
     let rawLibReferenceDirectives: readonly FileReference[];
     const resolver = context.getEmitResolver();
     const options = context.getCompilerOptions();
-    const { stripInternal } = options;
+    const { stripInternal, isolatedDeclarations } = options;
     return transformRoot;
 
+    function reportExpandoFunctionErrors(node: FunctionDeclaration | VariableDeclaration) {
+        resolver.getPropertiesOfContainerFunction(node).forEach(p => {
+            if (isExpandoPropertyDeclaration(p.valueDeclaration)) {
+                const errorTarget = isBinaryExpression(p.valueDeclaration) ?
+                    p.valueDeclaration.left :
+                    p.valueDeclaration;
+
+                context.addDiagnostic(createDiagnosticForNode(

```

Comment:
Likewise, this should be within `checkFunctionDeclarationDiagnostics` and `checkVariableDeclaration` in the checker instead of during emit, since it doesn't block emit (and isn't reliant on information only available during emit like node-builder triggered errors).